src/bundles/robot_simulation/controllers/utils/mergeConfig.ts

```diff
@@ -0,0 +1,12 @@
+import * as _ from 'lodash';
+
+export type RecursivePartial<T> = T extends object ? {
+  [P in keyof T]?: RecursivePartial<T[P]>;
+} : T;
```

Comment:
Can we move this to common utils and perhaps name it `DeepPartial` (to match frontend)?

Reply:
Fixed!